checkpoint-labs/checkpoint
# feat: add codegen to generate ORM models from schema

Closes: https://github.com/checkpoint-labs/checkpoint/issues/205
Depends on: https://github.com/checkpoint-labs/checkpoint/pull/230

This PR adds very primitive codegen for our ORM. This is exposed with `checkpoint codegen` script.

**BREAKING CHANGE: I had to move `decimalTypes` from runtime opts to CheckpointConfig as it needs to be defined statically so we can use it in scripts.**

## Test plan
- Build checkpoint.
- Go to some repo (for example [sx-api](https://github.com/snapshot-labs/sx-api/compare/sekhmet/codegen?expand=1) - this branch is converted to use ORM) that uses checkpoint and link checkpoint.
- Run `node path/to/checkpoint/dist/src/bin/index.js generate`.
- `.checkpoint/models.ts` should be generated.
- Try using models to interact with DB - if using sx-api from above repo is already converted to use this ORM.
